File: Ejercicio personajes (Fabrica abstracta)/Fabrica abstracta V3/src/Fabricas/FabricaElfos.java
package Fabricas;

import Personajes.ArmaduraElfo;
import Personajes.ArmaElfo;
import Personajes.CuerpoElfo;
import Personajes.MonturaElfo;
import Interfaces.IArmadura;
import Interfaces.IArma;
import Interfaces.ICuerpo;
import Interfaces.IMontura;

public class FabricaElfos extends FabricaAbstracta {

    @Override
    public IArma crearArma() {
        return new ArmaElfo();
    }

    @Override
    public IArmadura crearArmadura() {
        return new ArmaduraElfo();
    }

    @Override
    public IMontura crearMontura() {
        return new MonturaElfo();
    }

    @Override
    public ICuerpo crearCuerpo() {
        return new CuerpoElfo();
    }
}

File: Ejercicio personajes (Fabrica abstracta)/CombatChoose/CombatChoose/fabricas/fabrica_elfos.py
from fabricas.fabrica_abstracta import FabricaAbstracta
from interfaces.arma import IArma
from interfaces.armadura import IArmadura
from interfaces.cuerpo import ICuerpo
from interfaces.montura import IMontura

class Arco(IArma):
    def get_descripcion(self):
        return "Arco largo élfico"

class ArmaduraLigera(IArmadura):
    def get_descripcion(self):
        return "Armaduras de cuero ligero"

class CuerpoElfo(ICuerpo):
    def get_descripcion(self):
        return "Ágil y elegante"

class Ciervo(IMontura):
    def get_descripcion(self):
        return "Ciervo místico"

class FabricaElfos(FabricaAbstracta):
    def crear_arma(self):
        return Arco()
    def crear_armadura(self):
        return ArmaduraLigera()
    def crear_cuerpo(self):
        return CuerpoElfo()
    def crear_montura(self):
        return Ciervo()

